### ISSUE TITLE
Is it safe to update hdb_catalog.hdb_permission outside of Hasura?

### ISSUE THREAD

**Issue by raarts** [hasura/graphql-engine#5000]
Hi,

I'd like to put my permissions in version control. 
Is it allowed to just modify the hdb_catalog.hdb_permission table? Is Hasura subscribed to that table, i.e. will it pick up my changes automatically?

Thanks.

**Comment by rikinsk** [comment]
Yes, this should be possible. Though it is recommended that you use [hasura migrations](https://hasura.io/docs/1.0/graphql/manual/migrations/index.html) to version control your hasura metadata.

In any case, updating the `hdb_permission` table directly should also work. You might have to reload the metadata after you make your changes to apply them by any of the methods shown [here](https://hasura.io/docs/1.0/graphql/manual/migrations/manage-metadata.html#reloading-hasura-metadata)

**Comment by raarts** [comment]
Thanks for pointing me to the migrations feature. Completely missed that.

**Comment by raarts** [comment]
Hi. After looking at it, it looks like the migrations feature is not working for my workflow.
My workflow:

1. I use a Gitlab CI/CD pipeline, a docker swarm for deploy, and dev laptop with docker
2. I write multiple microservices (some of which are: postgres, hasura, backend, frontend)
3. When I write a new feature, go to the backend directory and in the migrations directory I add an SQL script that adds some column or table.
4. Commit, and the pipeline builds a docker image which includes the migration directory (because I cannot mount it into the container at runtime, I don't know on which node the swarm will run the container)
5. Deploy, and the backend at startup reads all migration files, and applies the last one

Now I want to apply hasura permissions to this workflow. But I don't see how this works in your migrations feature, since hasura-cli does not go through a list of migration files, and applies the last one, both not for the SQL files as the YAML files.

This means I cannot go into my migration directories and search older migrations, for examples how I did things, or which permissions actually apply to which commits, since each new hasura docker image I build, only contains the last migration and metadata status.

I hope I explained my usecase well. How do I proceed? Thanks.
